src/qibolab/_core/instruments/qm/config/devices.py

```diff
@@ -53,29 +75,37 @@ def from_config(cls, config: QmAcquisitionConfig):
 class OctaveOutput:
     LO_frequency: int
     gain: int = 0
-    LO_source: str = "internal"
-    output_mode: str = "triggered"
+    LO_source: Literal["internal", "external"] = "internal"
+    output_mode: OctaveOutputModes = "triggered"
 
     @classmethod
-    def from_config(cls, config: OscillatorConfig):
-        return cls(LO_frequency=config.frequency, gain=config.power)
+    def from_config(cls, config: Union[OscillatorConfig, OctaveOscillatorConfig]):
+        kwargs = dict(LO_frequency=config.frequency, gain=config.power)
+        if isinstance(config, OctaveOscillatorConfig):
+            kwargs["output_mode"] = config.output_mode
+        return cls(**kwargs)
 
 
 @dataclass(frozen=True)
 class OctaveInput:
     LO_frequency: int
-    LO_source: str = "internal"
-    IF_mode_I: str = "direct"
-    IF_mode_Q: str = "direct"
+    LO_source: Literal["internal", "external"] = "internal"
+    IF_mode_I: Literal["direct", "envelop", "mixer"] = "direct"
+    IF_mode_Q: Literal["direct", "envelop", "mixer"] = "direct"
+
+
+ModuleTypes = Literal["opx1", "LF", "MW"]
 
 
 @dataclass
 class Controller:
+    type: ModuleTypes = "opx1"
+    """https://docs.quantum-machines.co/latest/docs/Introduction/config/?h=opx10#controllers"""
     analog_outputs: PortDict[dict[str, AnalogOutput]] = field(default_factory=PortDict)
     digital_outputs: PortDict[dict[str, dict]] = field(default_factory=PortDict)
-    analog_inputs: PortDict[dict[str, AnalogInput]] = field(
-        default_factory=lambda: PortDict(DEFAULT_INPUTS)
-    )
+    analog_inputs: PortDict[dict[str, AnalogInput]] = field(default_factory=PortDict)
+    # default_factory=lambda: PortDict(DEFAULT_INPUTS)
+    # )
```

Comment:
Dead code?

Reply:
Thanks for catching this. I decided to go with the original one (that is: not change in this PR), as I think the default value there was needed for something related to the mixer calibration. I just had to also add default offsets (see e4947d94c3d49ffb15f870a7be5acf7f046888e2) because qm-qua>1.2 fails without them.